viewer/src/utilities/workers/WorkerPool.ts

```diff
@@ -74,15 +75,22 @@ export class WorkerPool {
   }
 
   async postWorkToAvailable<T>(work: WorkDelegate<T>): Promise<T> {
-    let targetWorker = this.workerList[0];
-    for (const worker of this.workerList) {
-      if (worker.activeJobCount < targetWorker.activeJobCount) {
-        targetWorker = worker;
+    const targetWorker = this.workerList.reduce((bestWorker, candidate) => {
+      if (bestWorker === undefined || bestWorker.activeJobCount > candidate.activeJobCount) {
```

Comment:
To be honest I don't understand what's the difference here between the previous form

```
    let targetWorker = this.workerList[0];
    for (const worker of this.workerList) {
      if (worker.activeJobCount < targetWorker.activeJobCount) {
        targetWorker = worker;
      }
    }
```

and the current one. In both, you just iterate over all workers and reasign the chosen one. Though I like your reduce form more. 

---

Also, I don't understand why `bestWorker === undefined` check is needed? You initialised it with this.workerList[0] already, so it can't be undefined?

Reply:
You are right - I misread the code and thought it would always pick the first worker when there was a queue.